synapse/api/filtering.py

```diff
@@ -198,7 +198,9 @@ def check(self, event):
         sender = event.get("sender", None)
         if not sender:
             # Presence events have their 'sender' in content.user_id
-            sender = event.get("content", {}).get("user_id", None)
+            content = event.get("content")
+            if isinstance(content, dict):
+                sender = content.get("user_id")
```

Comment:
Could you add a brief comment along the lines of: "Synapse has allowed account_data to have a non-dict content, so lets not assume it here", just to make sure no one changes it back in the future if we start enforcing account data that `content` is a `dict`